app/script/calling/entities/Flow.coffee

```diff
@@ -757,33 +760,36 @@ class z.calling.entities.Flow
     @audio.inject_audio_file audio_file_path, callback
 
   ###
-  Switch out the local MediaStream.
+  Update the local MediaStream.
   @param media_stream_info [z.calling.payloads.MediaStreamInfo] Object containing the required MediaStream information
   @return [Promise] Promise that resolves when the updated MediaStream is used
   ###
-  switch_media_stream: (media_stream_info) =>
-    if @peer_connection.getSenders?
-      @_replace_media_track media_stream_info
-      .then ->
-        return [media_stream_info, false]
-    else
-      @_replace_media_stream media_stream_info
-      .then (media_stream_info) =>
-        @is_answer false
-        return [media_stream_info, true]
+  update_media_stream: (media_stream_info) =>
+    @_replace_media_track media_stream_info
+    .catch (error) =>
+      if error.type is z.calling.CallError::TYPE.NOT_SUPPORTED
+        @logger.log @logger.levels.INFO, "Replacement of MediaStream and renegotiation necessary: #{error.message}", error
+        return @_replace_media_stream media_stream_info
+      else
+        throw error
```

Comment:
no need for `else` since there is `return` right before it.. Just the `throw error` is enough..